### ISSUE TITLE
Publish GitHub releases with new version of `pycromanager`

### ISSUE THREAD

**Issue by ieivanov** [micro-manager/pycro-manager#611]
Currently, it is hard to keep track of how `pycromanager` is evolving with increasing version numbers. When upgrading to the latest version I always wonder how it is different from the one I currently have - what new features and bug fixes does it include?

One way to address this issue is using GitHub releases which accompany pypi releases. I think `napari` provides a good example for this: https://github.com/napari/napari/releases. As I understand it, GitHub releases can be automatically generated on, say, major and minor version increments of the software. Reports on new features and bug fixes can also be automatically generated based on tags in the PRs. The GitHub releases can then be used to compare different versions of the software.

I don't know how to do this myself, but I'm sure we can figure it out if we agree it is valuable. We may be able to get help on this from @ziw-liu

Related issue: https://github.com/micro-manager/NDTiffStorage/issues/106

**Comment by henrypinkard** [comment]
Yes I think this is a good idea. Having a record of which pypi versions were released when would also be good for the purposes of matching with nightly builds
